week1/list1_done.py

```diff
@@ -0,0 +1,111 @@
+from operator import itemgetter
+# Given a list of strings, return the count of the number of
+# strings where the string length is 2 or more and the first
+# and last chars of the string are the same.
+# Note: python does not have a ++ operator, but += works.
+
+
+def match_ends_with_lambda(words):
+    length_list = list(filter(lambda x: len(x) >= 2, words))
+    list_to_return = list(filter(lambda x: x[0] == x[-1], length_list))
+    return len(list_to_return)
+
+
+def match_ends_without_lambda(words):
+    list_to_return = []
+    for word in words:
+        if (len(word) >= 2) and (word[0] == word[-1]):
+            list_to_return.append(word)
+    return len(list_to_return)
+
+
+def my_test(got, expected):
+    if got == expected:
+        print("Pass")
+    else:
+        print("Fail")
+
+# Simple provided test() function used in main() to print
+# what each function returns vs. what it's supposed to return.
+
+
+def test(got, expected):
+    if got == expected:
+        prefix = ' OK '
+    else:
+        prefix = '  X '
+    print('%s got: %s expected: %s' % (prefix, repr(got), repr(expected)))
+
+# Given a list of strings, return a list with the strings
+# in sorted order, except group all the strings that begin with 'x' first.
+# e.g. ['mix', 'xyz', 'apple', 'xanadu', 'aardvark'] yields
+# ['xanadu', 'xyz', 'aardvark', 'apple', 'mix']
+# Hint: this can be done by making 2 lists and sorting each of them
+# before combining them.
+
+
+def front_x_with_lambda(words):
+    list_with_x = list(filter(lambda x: (x.startswith("x")), words))
+    other_list = list(filter(lambda x: (not x.startswith("x")), words))
```

Comment:
Usually simple lambda functions are written without parentheses around its body. (And for something more complex it's better to use nested function instead of a lambda).
Please remove parentheses.